pkg/status/message/message_service.go

```diff
@@ -0,0 +1,142 @@
+func NewStatusMessageContext(pluginName, documentationSection string, pr *gogh.PullRequest, config config.PluginConfiguration) StatusMessageContext {
+	return StatusMessageContext{
+		pluginName:           pluginName,
+		documentationSection: documentationSection,
+		pullRequest:          pr,
+		config:               config,
+	}
+}
+
+// NewStatusMessageService creates an instance of GitHub StatusMessageService for the given StatusMessageContext
+func NewStatusMessageService(client ghclient.Client, log log.Logger, commentsLoader *ghservice.IssueCommentsLazyLoader, commentContext StatusMessageContext) *StatusMessageService {
+	return &StatusMessageService{
+		commentService: &ghservice.CommentService{
+			Client: client,
+			Issue:  commentsLoader.Issue,
+		},
+		log:            log,
+		commentContext: commentContext,
+		commentsLoader: commentsLoader,
+		change:         ghservice.NewRepositoryChangeForPR(commentContext.pullRequest),
+	}
+}
+
+// SadStatusMessage creates a message with the sad Ike image
+func (s *StatusMessageService) SadStatusMessage(description, statusFileSpec string, addIfMissing bool) error {
+	return s.StatusMessage(func() string {
+		messageLoader := s.newMessageLoader(sadIke, description)
+		return messageLoader.LoadMessage(s.change, statusFileSpec)
+	}, addIfMissing)
+}
+
+// HappyStatusMessage creates a message with the happy Ike image
+func (s *StatusMessageService) HappyStatusMessage(description, statusFileSpec string, addIfMissing bool) error {
+	return s.StatusMessage(func() string {
+		messageLoader := s.newMessageLoader(happyIke, description)
+		return messageLoader.LoadMessage(s.change, statusFileSpec)
+	}, addIfMissing)
+}
+
+func (s *StatusMessageService) newMessageLoader(image, msg string) *Loader {
+	return &Loader{
+		Log:        s.log,
+		PluginName: s.commentContext.pluginName,
+		Message: &Message{
+			Thumbnail:     image,
+			Description:   msg,
+			ConfigFile:    s.commentContext.config.LocationURL,
+			Documentation: s.commentContext.documentationSection,
+		},
+	}
+}
+
+// StatusMessage checks all present comments in the issue/pull-request. If no comment with PluginTitleTemplate
+// (with the related plugin) is found, then it adds a new comment with the plugin title, assignee mention
+// and the given commentMsg. If such a comment is present already, then it does nothing.
+func (s *StatusMessageService) StatusMessage(commentMsgCreator func() string, addIfMissing bool) error {
+	comments, err := s.commentsLoader.Load()
+	if err != nil {
+		s.log.Errorf("Getting all comments failed with an error: %s", err)
+	}
+	statusMsg := utils.String("")
+
+	for _, com := range comments {
+		content := *com.Body
+		if strings.HasPrefix(content, s.getPluginTitle()) {
+			s.loadStatusMessage(commentMsgCreator, statusMsg)
+			if strings.TrimSpace(content) == strings.TrimSpace(*statusMsg) {
+				return nil
+			}
+			return s.commentService.EditComment(*com.ID, statusMsg)
```

Comment:
I'm not sure we should edit comments. I look at PR comments (and events) as a timeline so I would rather append new comments with the relevant message. Maybe we can extend the message to point to commit? "It seems that prior to commit 6fas341213ad3a there are no tests in this PR" or sth like - to get the idea, as the wording is not ideal. WDYT?